client/ayon_blender/plugins/create/create_usd.py

```diff
@@ -20,11 +22,46 @@ def create(
             product_name, instance_data, pre_create_data
         )
 
+        objects = []
         if pre_create_data.get("use_selection"):
             objects = lib.get_selection()
+
+        # Create template hierarchy
+        if pre_create_data.get("createAssetTemplateHierarchy", False):
+            folder_path = instance_data["folderPath"]
+            folder_name = folder_path.rsplit("/", 1)[-1]
+
+            root = bpy.data.objects.new(folder_name, object_data=None)
+            bpy.context.scene.collection.objects.link(root)
+
+            geo = bpy.data.objects.new("geo", object_data=None)
+            bpy.context.scene.collection.objects.link(geo)
+            geo.parent = root
+
+            # Parent members with geo.
             for obj in objects:
-                collection.objects.link(obj)
-                if obj.type == 'EMPTY':
-                    objects.extend(obj.children)
+                obj.parent = geo
```

Comment:
This would go quite bad with @moonyuet's changes being made here: https://github.com/ynput/ayon-blender/pull/128

Because then suddenly, all children under your current selection would then also get reparented to this root - which is not what we'd want. So we may need to handle this case separately @moonyuet @MustafaJafar

Reply:
Only go quite bad when the condition of `checking the type of object is empty` is not removed once my PR is merged.

Reply:
That won't solve it - a hierarchy can exist of multiple empties parented to each other, that hierarchy should be maintained.

Reply:
With @moonyuet 's PR the easiest fix is to just NOT get the selection with hierarchy:

```python
selection = lib.get_selection()  # assuming this defaults to NOT including the unselected children hierarchy
for obj in selection:
   obj.parent = root

selection_hierarchy = set(selection) | lib.get_object_children_recursive(selection)
for obj in selection_hierarchy:
    collection.link(obj)
```

This is the most sensible approach here.